ignite/contrib/handlers/base_logger.py

```diff
@@ -105,6 +106,26 @@ def _setup_output_metrics(self, engine: Engine) -> Dict[str, Any]:
             metrics.update({name: value for name, value in output_dict.items()})
         return metrics
 
+    def _setup_state_attributes(self, engine: Engine) -> Dict[str, Any]:
+        """Helper method to setup state attributes to log"""
+        state_attributes_dict = OrderedDict()
+
+        if self.state_attributes is not None:
+            for name in self.state_attributes:
+
+                value = getattr(engine.state, name, None)
+                if value is None:
+                    continue
+                if isinstance(value, numbers.Number):
+                    state_attributes_dict[name] = value
+                if isinstance(value, torch.Tensor) and value.ndimension() == 0:
+                    state_attributes_dict[name] = value.item()
+                if isinstance(value, torch.Tensor) and value.ndimension() == 1:
+                    for i, v in enumerate(value):
+                        state_attributes_dict[f"{name}/{i}"] = v.item()
```

Comment:
I like very much this idea to handle data type here in the base class instead of derived class what we are doing today. Could you please send another PR to update `_setup_output_metrics` and perform the same logic there and thus we can remove data type dispatch in derived classes like TensorboardLogger etc ?
Then, I think here in this PR, we can rename `_setup_output_metrics` to `_setup_output_metrics_state_attrs` and handle everything in the same way. What do you think ?

Reply:
Yes I agree, this way all the other tracking system functions don't need to have the step repeated.